File: Problem/Easy/Problem_10/Solution.cpp
// C++ program using tail recursion to calculate sum of array elements.

#include <iostream>

// Recursive function
long long sumArray(int a[], int n = 0, int total = 0)
{
    if (n <= 0)
        return total;
    return sumArray(a, n - 1, total + a[n - 1]);
}

// Main function
int main()
{
    int n = 4;
    int arr[4] = {15, 12, 13, 10};
    std::cout << "Sum of array is: " << sumArray(arr, n);
    return 0;
}

File: Problem/Easy/Problem_10/Solution.py
# Python3 program using tail recursion to calculate sum of array elements.

# Recursive function
def sumOfArray(arr : list, n : int, total = 0) -> int:
    if n <= 0:
        return total
    return sumOfArray(arr, n - 1, total + arr[n - 1]);

# Main
if __name__ == "__main__":
    arr = [15, 12, 13, 10]
    print('Sum of array is : {}'.format(sumOfArray(arr, len(arr))))